sycl/doc/extensions/GroupAlgorithms/SYCL_INTEL_group_sort.asciidoc

```diff
@@ -175,44 +202,74 @@ Only arithmetic types as `T` can be passed to `radix_sorter`.
 `BitsPerPass` is a number of bits that values are split by.
 For example, if a sequence of `int32_t` is sorted using `BitsPerPass == 4` then one
 pass of the radix sort algorithm considers only 4 bits. The number of passes is `32/4=8`.
+The algorithm requires a temporary local memory that must be allocated on callers side.
+Size of required memory (bytes) is defined by calling `memory_required`.
 |===
 
-Table. Constructors of the `default_sorter` class.
+Table 4. Constructors of the `default_sorter` class.
 |===
 |Constructor|Description
 
-|`default_sorter(Compare comp = Compare())`
+|`template<std::size_t Extent>
+default_sorter(sycl::span<uint8_t, Extent> scratch, Compare comp = Compare())`
 |Creates the `default_sorter` object using `comp`.
+Temporary local memory for the algorithm is provided using `scratch`.
+If `scratch.size()` is less than value returned by
+`memory_required`, behavior of the corresponding sorting algorithm is undefined.
+
 |===
 
-Table. Member functions of the `default_sorter` class.
+Table 5. Member functions of the `default_sorter` class.
 |===
 |Member function|Description
 
 |`template<typename Group, typename Ptr>
 void operator()(Group g, Ptr first, Ptr last)`
 |Implements a default sorting algorithm to be called by the `joint_sort` algorithm.
 
-_Complexity_: Let `N` be `last - first`. `O(N*log_2(N))` comparisons.
+_Complexity_: Let `N` be `last - first`. `O(N*log(N)*log(N))` comparisons.
 
 |`template<typename Group, typename T>
 T operator()(Group g, T val)`
 |Implements a default sorting algorithm to be called by the `sort_over_group` algorithm.
 
-_Complexity_: Let `N` be the work group size. `O(N*log_2(N))` comparisons.
+_Complexity_: Let `N` be the work group size. `O(N*log(N)*log(N))` comparisons.
+|`template<int dimensions = 1>
+static std::size_t memory_required(sycl::memory_scope scope, std::size_t range_size)`
+|Returns size of temporary memory (in bytes) that is required by
+the default sorting algorithm defined by the sorter calling by `joint_sort`.
+`range_size` represents a range size for sorting,
+e.g. `last-first` from `operator()` arguments.
+Result depends on the `scope` parameter:
+use `sycl::memory_scope::work_group` if you want to run algorithms using `sycl::group`;
+use `sycl::memory_scope::sub_group` if you want to run algorithms using `sycl::sub_group`.
+Function must be called on host.

```

Comment:
Where does this restriction come from?

Reply:
The thing is memory_required function is needed to get memory size for allocating memory. We can allocate additional local memory only on host. So, calling the function on device is useless. Do you think this sentence is extra?

Reply:
Ok.  I think this sentence is unnecessary, then.  I can think of a few use-cases for this functionality in device code:
1) Reacting to the size of some scratch allocation -- if an allocation is big enough, run the built-in sort, otherwise do something else
2) Enabling smart users -- it's already possible to emulate memory allocation within local or global memory by implementing your own stack or heap on top of atomics
3) Enabling other (future) extensions -- if an extension introduces a mechanism to allocate memory within a kernel (which may be supported on some devices), they shouldn't have to modify this extension too